Source/player.cpp

```diff
@@ -1084,6 +1084,7 @@ void InitPlayer(int pnum, BOOL FirstTime)
 
 		if (pnum == myplr) {
 			plr[pnum]._plid = AddLight(plr[pnum]._px, plr[pnum]._py, plr[pnum]._pLightRad);
+			ChangeLightXY(plr[myplr]._plid, plr[myplr]._px, plr[myplr]._py); // fix for a bug where old light is still visible at the entrance after reentering level

```

Comment:
I have opted to fix this in `UpdateMonsterLights()` as that function can screw up light in existing saves and was called after `InitPlayer()`.

Reply:
And it's reverted with your fix and a better one for handling the monsters with incorrect light id in old saves :) https://github.com/diasurgical/devilutionX